src/chunk.c

```diff
@@ -1051,6 +1186,183 @@ chunk_find_all_oids(Hyperspace *hs, List *dimension_vecs, LOCKMODE lockmode)
+	if (PG_ARGISNULL(0))
+	{
+		hypertables = hypertable_get_all();
+	}
+	else
+	{
+		ht = hypertable_cache_get_entry(hypertable_cache, table_relid);
+		if (!ht)
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					 errmsg("table \"%s\" does not exist or is not a hypertable",
+							get_rel_name(table_relid))
+					 ));
+		hypertables = list_make1(ht);
+	}
+
+	oldcontext = MemoryContextSwitchTo(funcctx->multi_call_memory_ctx);
+	ctxs = palloc(sizeof(ChunkScanCtx *) * list_length(hypertables));
+	MemoryContextSwitchTo(oldcontext);
+	foreach(lc, hypertables)
+	{
+		ht = lfirst(lc);
+
+		time_dim = hyperspace_get_open_dimension(ht->space, 0);
+
+		if (time_dim_type == InvalidOid)
+			time_dim_type = time_dim->fd.column_type;
+
+		/*
+		 * Even though internally all time columns are represented as bigints,
+		 * it is locally unclear what set of chunks should be returned if
+		 * there are multiple tables on the system some of which care about
+		 * timestamp when others do not. That is why, whenever there is any
+		 * time dimension constraint given as an argument (older_than or
+		 * newer_than) we make sure all hypertables have the time dimension
+		 * type of the given type or through an error. This check is done
+		 * accross hypertables that is why it is not in the helper function
+		 * below.
+		 */
+		if (time_dim_type != time_dim->fd.column_type &&
+			(older_than_type != InvalidOid || newer_than_type != InvalidOid))
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					 errmsg("cannot call \"%s\" on all hypertables "
+							"when all hypertables do not have the same time dimension type",
+							caller_name)
+					 ));
+
+		ctxs[ht_index++] = chunks_typecheck_and_find_all_in_range_limit(ht->space,
+																		time_dim,
+																		older_than_datum,
+																		older_than_type,
+																		newer_than_datum,
+																		newer_than_type,
+																		-1,
+																		funcctx->multi_call_memory_ctx,
+																		caller_name,
+																		&num_chunks);
+
+	}
+	oldcontext = MemoryContextSwitchTo(funcctx->multi_call_memory_ctx);
+	/* Q:: two options here: */
+	/* have chunks to be chunks and copy chunk structs over */
+	/* have chunks be Oids or a new lighter datatype */
+	/* First one has more overhead but makes ordering the results much easier */
+	/* as we can use hypertable d and table id. */
+
+	/*
+	 * The first one is more effient and makes sure we do not deal with mem
+	 * context switches that much
+	 */
+
+	/*
+	 * num_chunks can safely be 0 as palloc protects against unportable
+	 * behavior.
+	 */
+	chunks = palloc(sizeof(Chunk *) * num_chunks);
+	current = chunks;
+
+	MemoryContextSwitchTo(oldcontext);
+	for (int i = 0; i < list_length(hypertables); i++)
+	{
+		/* Get all the chunks from the context */
+		ctxs[i]->data = current;
+		chunk_scan_ctx_foreach_chunk(ctxs[i], chunk_scan_context_add_chunk, -1);
```

Comment:
hypertables already come in sorted order so we could technically do this instead of sorting everything later 
`qsort(current, (Chunk **) ctxs[i]->data - current, sizeof(Chunk *), chunk_cmp);`

Reply:
Not sure it is worth it. Also silently breaks if someone changes ordering of hypertables.

Reply:
Fwiw, if you'd do a partial sort, I a further optimization is probably to combine that with a lazy scan of chunks. So, on firstcall you just scan for hypertables, then in `chunks_return_srf` you lazily scan for chunks for each hypertable as needed. This would reduce memory usage as you only keep one per-hypertable chunks array at a  time and you need not scan some chunks in case of a LIMIT clause.

Just suggesting as feedback, but I don't think it is required that you do that now.